### DISCUSSION TITLE
Digital products

### DISCUSSION THREAD

**Discussion by mraf** [givanz/Vvveb#179] [General]
Hello, 

is there roadmap for support for digital products maybe? 

I know @givanz you mentioned some time ago about multi vendor support for it which is great. Below my thoughts about other things:

1. Digital products media items stored on local ( like digital assest with all security ) or served from s3 or minio endpoint (that requires plugin for sure).
2. Manage digital products from the product dashboard ( looks like assigning digital asset on product tab is disabled now ).
3. Handle and fulfill digital product orders.
4. Allow customers to download their digital product purchases from the storefront. ( already there with digital assets -> Digital assets assigned to a product show in the user > download section after they buy the product. )

**Comment by givanz** [comment]
Hi

I finished adding missing features for digital assets in the last update.

* If the cart does not have product that require shipping (require shipping set to false on product page) then checkout will not display shipping method and shipping address fields
* Fixed autocomplete issues with digital assets in admin > product > relations
* Display downloads in user dashboard > downloads and allow secure file download, files will show only for orders that have status = `complete`

Files are stored in `/storage/digital_assets` with recommended secure install https://docs.vvveb.com/installation#secure-install files are secure and can't be downloaded in any other way without purchasing.

I will add necessary events/filters to allow plugins to override storage and download to serve files from S3 along with download log view in a next update. 

Multi vendor will be finished after advanced product filters and subscriptions.

Let me know if have any issues or feedback with product downloads.

Thanks

**Comment by mraf** [comment]
Brilliant news! I will get latest and will let you know how I managed to make it working
